src/utils/hooks/useGetFullApplicationStructure.tsx

```diff
@@ -84,10 +92,28 @@ const useGetFullApplicationStructure = ({
       if (shouldDoValidation) {
         newStructure.lastValidationTimestamp = Date.now()
       }
-      if (shouldCalculateProgress) generateResponsesProgress(newStructure)
+
+      newStructure = addElementsById(newStructure)
+      newStructure = addSortedSectionsAndPages(newStructure)
+
+      if (shouldCalculateProgress) {
+        // even thought response is already added to element, we need to Latest and Previous applicaiton response
+        addApplicationResponses(newStructure, applicationResponses)
+        addApplicantChangeRequestStatusToElement(newStructure)
+
+        generateApplicantChangesRequestedProgress(newStructure)
+
+        // generateResponseProgress uses change statuses calculated in generateApplicantChangesRequestedProgress
```

Comment:
```suggestion
        // generateResponseProgress uses isChanged statuses calculated in generateApplicantChangesRequestedProgress
```

Reply:
I'm not entirely sure I know what does this comment means

Reply:
generateReponseProgres now uses isChangeRequest and isChanged to apply validity and incompleteness, thus generateApplicantChangesRequestedProgress must be called first